src/Native/icon_win32.c

```diff
@@ -31,4 +34,85 @@ void revery_hideIconProgress_win32(void *win, void *ih) {
     iconHandle->lpVtbl->SetProgressState(iconHandle, window, TBPF_NOPROGRESS);
 }
 
+void revery_setIconBadge_win32(void *win, void *ih, char *badgeStr) {
+    HWND window = (HWND)win;
+    ITaskbarList3 *iconHandle = (ITaskbarList3 *)ih;
+
+    HDC hDC = GetDC(NULL);
+
+    // Load the badge template into memory
+    HBITMAP bmp = CreateCompatibleBitmap(hDC, 32, 32);
+
+    RECT colorRect;
+    colorRect.left = colorRect.top = 0;
+    colorRect.right = 32;
+    colorRect.bottom = 32;
+    HBRUSH hBrush = CreateSolidBrush(RGB(238, 17, 17));
+
+    int fontSize;
+
+    switch (strlen(badgeStr)) {
+    case 1:
+        fontSize = 30;
+        break;
+    case 2:
+        fontSize = 24;
+        break;
+    default:
+        fontSize = 20;
+        break;
+    }
+
+
+
+    // Create a rect to paint the text into on top of the bitmap
+    RECT textRect;
+    textRect.left = 0;
```

Comment:
```suggestion
    textRect.left = (32 - fontSize) / 2;
```
So it aligned vertically I don't see where we specify line height.

Reply:
It already is aligned vertically. Check out `DrawText`, i.e. `DT_CENTER` and `DT_VCENTER`.

Line height is also specified as the first argument to `DrawText`

Reply:
`memHDC` ?

Reply:
My mistake 😳, it is the first argument to `CreateFont`:
https://github.com/revery-ui/revery/blob/3c1cf15f02243e8ea41fd75fbdb7e709f9b395c4/src/Native/icon_win32.c#L83-L85